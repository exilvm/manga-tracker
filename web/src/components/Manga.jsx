import React, { useCallback, useState } from 'react';
import {
  Button,
  Container,
  Grid, IconButton,
  Paper,
  Tooltip,
  Typography,
} from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';

import { Edit as EditIcon } from '@material-ui/icons';
import MangaSourceList from './MangaSourceList';
import {
  defaultDateDistanceToNow,
  defaultDateFormat,
  followUnfollow,
} from '../utils/utilities';
import ChapterList from './ChapterList';
import { useUser } from '../utils/useUser';


const useStyles = makeStyles((theme) => ({
  title: {
    width: '75%',
    textAlign: 'left',
    paddingBottom: '10px',
    [theme.breakpoints.down('sm')]: {
      width: '100%',
    },
  },
  titleBar: {
    display: 'flex',
    justifyContent: 'space-between',
  },
  thumbnail: {
    maxWidth: '250px',
    maxHeight: '355px',
    [theme.breakpoints.down('sm')]: {
      maxWidth: '180px',
    },
    [theme.breakpoints.down('xs')]: {
      maxWidth: '125px',
    },
  },
  details: {
    display: 'flex',
  },
  detailText: {
    marginLeft: '5px',
    [theme.breakpoints.down('sm')]: {
      marginLeft: '3px',
    },
  },
  infoTable: {
    marginLeft: '30px',
    marginTop: '3px',
    [theme.breakpoints.down('sm')]: {
      marginLeft: '20px',
    },
    [theme.breakpoints.down('xs')]: {
      marginLeft: '10px',
    },
  },
  sourceList: {
    marginLeft: '35px',
    [theme.breakpoints.down('sm')]: {
      marginLeft: '23px',
    },
    [theme.breakpoints.down('xs')]: {
      marginLeft: '13px',
    },
  },
  followButton: {
    marginTop: theme.spacing(2),
    marginBottom: theme.spacing(2),
  },
  paper: {
    padding: '1em',
    minWidth: '440px',
  },
}));


function Manga(props) {
  const {
    mangaData,
    userFollows = [],
  } = props;

  const { isAuthenticated, user } = useUser();

  const classes = useStyles();
  const [editing, setEditing] = useState(false);
  const startEditing = useCallback(() => setEditing(!editing), [editing]);

  const latestRelease = mangaData.latest_release ?
    new Date(mangaData.latest_release) :
    null;

  const estimatedRelease = new Date(mangaData.estimated_release);

  const mangaChapters = React.useMemo(() => {
    if (!mangaData.chapters) return null;
    const serviceMap = {};
    mangaData.services.forEach(service => { serviceMap[service.service_id] = service.url_format });
    return mangaData.chapters.map(chapter => {
      const newChapter = { ...chapter };
      newChapter.release_date = new Date(chapter.release_date);
      newChapter.url = serviceMap[chapter.service_id].replace('{}', chapter.chapter_url);
      return newChapter;
    });
  }, [mangaData.chapters, mangaData.services]);

  return (
    <Container maxWidth='lg'>
      <Paper className={classes.paper}>
        <div className={classes.titleBar}>
          <Typography className={classes.title} variant='h4'>{mangaData.title}</Typography>
          {user?.admin && (
            <IconButton onClick={startEditing}>
              <EditIcon />
            </IconButton>
          )}
        </div>
        <div className={classes.details}>
          <a href={mangaData.mal} target='_blank' rel='noreferrer noopener'>
            <img
              src={mangaData.cover}
              className={classes.thumbnail}
              alt={mangaData.title}
            />
          </a>
          <Grid
            container
            justify='space-between'
          >
            <table className={classes.infoTable}>
              <tbody>
                <tr>
                  <td><Typography>Latest release:</Typography></td>
                  <td>
                    <Tooltip title={latestRelease ? latestRelease.toUTCString() : 'Unknown'}>
                      <Typography className={classes.detailText}>
                        {latestRelease ?
                          defaultDateFormat(latestRelease) + ' - ' + defaultDateDistanceToNow(latestRelease) :
                          'Unknown'}
                      </Typography>
                    </Tooltip>
                  </td>
                </tr>
                <tr>
                  <td><Typography>Estimated release interval:</Typography></td>
                  <td>
                    <Typography className={classes.detailText}>
                      {(mangaData.release_interval ?
                        `${mangaData.release_interval?.days || 0} days ${mangaData.release_interval?.hours || 0} hours` :
                        'Unknown')}
                    </Typography>
                  </td>
                </tr>
                <tr>
                  <td><Typography>Estimated next release:</Typography></td>
                  <td>
                    <Typography className={classes.detailText}>
                      {defaultDateFormat(estimatedRelease)}
                    </Typography>
                  </td>
                </tr>
                <tr>
                  <td><Typography>Latest chapter:</Typography></td>
                  <td>
                    <Typography className={classes.detailText}>
                      {mangaData.latest_chapter ? mangaData.latest_chapter : 'Unknown'}
                    </Typography>
                  </td>
                </tr>
              </tbody>
            </table>
            <MangaSourceList
              classesProp={[classes.sourceList]}
              items={mangaData.services}
              userFollows={userFollows}
              followUnfollow={(serviceId) => followUnfollow(mangaData.manga_id, serviceId)}
            />
          </Grid>
        </div>
        {isAuthenticated && (
          <Button
            variant='contained'
            color='primary'
            onClick={followUnfollow(mangaData.manga_id, null)}
            className={classes.followButton}
          >
            {userFollows.indexOf(null) < 0 ? 'Follow' : 'Unfollow'}
          </Button>
        )}

        <ChapterList
          chapters={mangaChapters}
          editable={editing}
        />
      </Paper>
    </Container>
  );
}

export default Manga;
