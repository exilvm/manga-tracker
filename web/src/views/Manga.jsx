import React from 'react';
import {
  Button,
  Container,
  Grid,
  Link,
  Paper,
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableRow,
  Tooltip,
  Typography,
} from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';

import MangaSourceList from '../components/MangaSourceList';
import { followUnfollow } from '../utils/utilities';


const useStyles = makeStyles((theme) => ({
  title: {
    width: '75%',
    textAlign: 'left',
    paddingBottom: '10px',
    [theme.breakpoints.down('sm')]: {
      width: '100%',
    },
  },
  thumbnail: {
    maxWidth: '250px',
    maxHeight: '355px',
    [theme.breakpoints.down('sm')]: {
      maxWidth: '180px',
    },
    [theme.breakpoints.down('xs')]: {
      maxWidth: '125px',
    },
  },
  details: {
    display: 'flex',
  },
  detailText: {
    marginLeft: '5px',
    [theme.breakpoints.down('sm')]: {
      marginLeft: '3px',
    },
  },
  infoTable: {
    marginLeft: '30px',
    marginTop: '3px',
    [theme.breakpoints.down('sm')]: {
      marginLeft: '20px',
    },
    [theme.breakpoints.down('xs')]: {
      marginLeft: '10px',
    },
  },
  sourceList: {
    marginLeft: '35px',
    [theme.breakpoints.down('sm')]: {
      marginLeft: '23px',
    },
    [theme.breakpoints.down('xs')]: {
      marginLeft: '13px',
    },
  },
  followButton: {
    marginTop: theme.spacing(2),
    marginBottom: theme.spacing(2),
  },
  paper: {
    padding: '1em',
    minWidth: '440px',
  },
  titleColumn: {
    minWidth: '225px',
  },
  releaseColumn: {
    minWidth: '200px',
  },
}));

const dateOptions = { weekday: 'short', year: 'numeric', month: 'long', day: 'numeric', hour: '2-digit', minute: '2-digit' };


function Manga(props) {
  const {
    mangaData,
    isAuthenticated = false,
    userFollows = [],
  } = props;

  const classes = useStyles();
  const latestRelease = mangaData.latest_release ?
      new Date(mangaData.latest_release) :
      null;

  const estimatedRelease = mangaData.estimated_release ?
    new Date(mangaData.estimated_release) :
    null;

  const mangaChapters = React.useMemo(() => {
    if (!mangaData.chapters) return null;
    const serviceMap = {};
    mangaData.services.forEach(service => { serviceMap[service.service_id] = service.url_format });
    return mangaData.chapters.map(chapter => {
      const newChapter = { ...chapter };
      newChapter.release_date = chapter.release_date ? new Date(chapter.release_date).toLocaleString('en-GB', dateOptions) : 'Unknown';
      newChapter.url = serviceMap[chapter.service_id].replace('{}', chapter.chapter_url);
      return newChapter;
    });
  }, [mangaData.chapters, mangaData.services]);

  return (
    <Container maxWidth='lg'>
      <Paper className={classes.paper}>
        <Typography className={classes.title} variant='h4'>{mangaData.title}</Typography>
        <div className={classes.details}>
          <a href={mangaData.mal} target='_blank' rel='noreferrer noopener'>
            <img
              src={mangaData.cover}
              className={classes.thumbnail}
              alt={mangaData.title}
            />
          </a>
          <Grid
            container
            justify='space-between'
          >
            <table className={classes.infoTable}>
              <tbody>
                <tr>
                  <td><Typography>Latest release:</Typography></td>
                  <td>
                    <Tooltip title={latestRelease ? latestRelease.toUTCString() : 'Unknown'}>
                      <Typography className={classes.detailText}>
                        {(latestRelease ? latestRelease.toLocaleString('en-GB', dateOptions) : 'Unknown')}
                      </Typography>
                    </Tooltip>
                  </td>
                </tr>
                <tr>
                  <td><Typography>Estimated release interval:</Typography></td>
                  <td>
                    <Typography className={classes.detailText}>
                      {(mangaData.release_interval ?
                          `${mangaData.release_interval?.days || 0} days ${mangaData.release_interval?.hours || 0} hours` :
                          'Unknown')}
                    </Typography>
                  </td>
                </tr>
                <tr>
                  <td><Typography>Estimated next release:</Typography></td>
                  <td>
                    <Typography className={classes.detailText}>
                      {estimatedRelease ? estimatedRelease.toLocaleString('en-GB', dateOptions) : 'Unknown'}
                    </Typography>
                  </td>
                </tr>
                <tr>
                  <td><Typography>Latest chapter:</Typography></td>
                  <td>
                    <Typography className={classes.detailText}>
                      {mangaData.latest_chapter ? mangaData.latest_chapter : 'Unknown'}
                    </Typography>
                  </td>
                </tr>
              </tbody>
            </table>
            <MangaSourceList
              classesProp={[classes.sourceList]}
              items={mangaData.services}
              userFollows={userFollows}
              isAuthenticated={isAuthenticated}
              followUnfollow={(serviceId) => followUnfollow(mangaData.manga_id, serviceId)}
            />
          </Grid>
        </div>
        {isAuthenticated && (
          <Button
            variant='contained'
            color='primary'
            onClick={followUnfollow(mangaData.manga_id, null)}
            className={classes.followButton}
          >
            {userFollows.indexOf(null) < 0 ? 'Follow' : 'Unfollow'}
          </Button>
        )}

        {mangaChapters && mangaChapters.length > 0 && (
        <TableContainer component={Paper}>
          <Table className={classes.table} aria-label='simple table'>
            <TableHead>
              <TableRow>
                <TableCell className={classes.titleColumn}>Title</TableCell>
                <TableCell>Chapter</TableCell>
                <TableCell className={classes.releaseColumn}>Released</TableCell>
                <TableCell>Group</TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {mangaChapters.map((row, index) => (
                // eslint-disable-next-line react/no-array-index-key
                <TableRow key={index}>
                  <TableCell component='th' scope='row'>
                    <Link href={row.url} target='_blank' style={{ textDecoration: 'none' }}>
                      <span>
                        {row.title}
                      </span>
                    </Link>
                  </TableCell>
                  <TableCell>{row.chapter_number}</TableCell>
                  <TableCell>{row.release_date}</TableCell>
                  <TableCell>{row.group}</TableCell>
                </TableRow>
              ))}
            </TableBody>
          </Table>
        </TableContainer>
      )}
      </Paper>
    </Container>
  );
}

export default Manga;
