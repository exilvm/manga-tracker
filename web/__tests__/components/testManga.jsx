import React from 'react';
import renderer from 'react-test-renderer';
import Manga from '../../src/views/Manga';

describe('Manga page should render correctly', () => {
  const manga = { manga_id: 1,
      title: 'Dr. STONE',
      release_interval: {
        days: 7,
      },
      latest_release: '2020-07-05T16:00:00.000Z',
      estimated_release: '2020-07-12T16:00:00.000Z',
      latest_chapter: 157,
      services: [
        {
          title_id: '100010',
          service_id: 1,
          name: 'MANGA Plus',
          url_format: 'https://mangaplus.shueisha.co.jp/viewer/{}',
          url: 'https://mangaplus.shueisha.co.jp/titles/{}',
        },
        {
          title_id: '20882',
          service_id: 2,
          name: 'MangaDex',
          url_format: 'https://mangadex.org/chapter/{}',
          url: 'https://mangadex.org/title/{}',
        },
      ],
      cover: 'https://mangadex.org/images/manga/20882.jpg?1585634146',
      status: 0,
      artist: 'Boichi',
      author: 'Inagaki Riichiro',
      last_updated: '2020-06-28T08:15:55.170Z',
      bw: 'https://bookwalker.jp/series/114645',
      mu: 'https://www.mangaupdates.com/series.html?id=139601',
      mal: 'https://myanimelist.net/manga/103897',
      amz: 'https://www.amazon.co.jp/gp/product/B075F8JBQ1',
      ebj: 'https://www.ebookjapan.jp/ebj/413780/',
      engtl: 'https://www.viz.com/dr-stone',
      raw: 'null',
      nu: 'https://www.novelupdates.com/series/null',
      kt: 'https://kitsu.io/manga/38860',
      ap: 'https://www.anime-planet.com/manga/dr-stone',
      al: 'https://anilist.co/manga/98416',
      chapters: [
        {
          title: 'Z=157: Same Time, Same Place',
          chapter_number: 157,
          release_date: 'Sun, July 5, 2020, 07:00 PM',
          group: 'Shueisha',
          service_id: 1,
          chapter_url: '1007322',
          url: 'https://mangaplus.shueisha.co.jp/viewer/1007322',
        },
        {
          title: 'Z=156: Two Scientists',
          chapter_number: 156,
          release_date: 'Fri, June 26, 2020, 08:01 PM',
          group: 'MangaPlus',
          service_id: 2,
          chapter_url: '938629',
          url: 'https://mangadex.org/chapter/938629',
        },
        {
          title: 'Z=156: Two Scientists',
          chapter_number: 156,
          release_date: 'Fri, June 26, 2020, 07:00 PM',
          group: 'Shueisha',
          service_id: 1,
          chapter_url: '1007024',
          url: 'https://mangaplus.shueisha.co.jp/viewer/1007024',
        },
        {
          title: 'Z=155: Science is Elegant',
          chapter_number: 155,
          release_date: 'Sun, June 21, 2020, 07:00 PM',
          group: 'Shueisha',
          service_id: 1,
          chapter_url: '1007023',
          url: 'https://mangaplus.shueisha.co.jp/viewer/1007023',
        },
      ],
    };
  const follows = [1];

  jest.spyOn(Date.prototype, 'toLocaleString')
    .mockImplementation(jest.fn(function toLocaleString() {
      return this.toUTCString();
    }));

  it('should render correctly', () => {
    const tree = renderer
      .create(<Manga mangaData={{ ...manga }} isAuthenticated={false} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when logged in', () => {
       const tree = renderer
      .create(<Manga mangaData={{ ...manga }} isAuthenticated follows={follows} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  })
});
