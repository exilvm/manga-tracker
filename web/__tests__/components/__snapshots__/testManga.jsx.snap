// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Manga page should render correctly should render correctly 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg"
>
  <div
    className="MuiPaper-root makeStyles-paper-8 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <h4
      className="MuiTypography-root makeStyles-title-1 MuiTypography-h4"
    >
      Dr. STONE
    </h4>
    <div
      className="makeStyles-details-3"
    >
      <a
        href="https://myanimelist.net/manga/103897"
        rel="noreferrer noopener"
        target="_blank"
      >
        <img
          alt="Dr. STONE"
          className="makeStyles-thumbnail-2"
          src="https://mangadex.org/images/manga/20882.jpg?1585634146"
        />
      </a>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
      >
        <table
          className="makeStyles-infoTable-5"
        >
          <tbody>
            <tr>
              <td>
                <p
                  className="MuiTypography-root MuiTypography-body1"
                >
                  Latest release:
                </p>
              </td>
              <td>
                <p
                  aria-describedby={null}
                  className="MuiTypography-root makeStyles-detailText-4 MuiTypography-body1"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  title="Sun, 05 Jul 2020 16:00:00 GMT"
                >
                  2020-07-05T16:00:00.000Z
                </p>
              </td>
            </tr>
            <tr>
              <td>
                <p
                  className="MuiTypography-root MuiTypography-body1"
                >
                  Estimated release interval:
                </p>
              </td>
              <td>
                <p
                  className="MuiTypography-root makeStyles-detailText-4 MuiTypography-body1"
                >
                  7 days 0 hours
                </p>
              </td>
            </tr>
            <tr>
              <td>
                <p
                  className="MuiTypography-root MuiTypography-body1"
                >
                  Estimated next release:
                </p>
              </td>
              <td>
                <p
                  className="MuiTypography-root makeStyles-detailText-4 MuiTypography-body1"
                >
                  2020-07-12T16:00:00.000Z
                </p>
              </td>
            </tr>
            <tr>
              <td>
                <p
                  className="MuiTypography-root MuiTypography-body1"
                >
                  Latest chapter:
                </p>
              </td>
              <td>
                <p
                  className="MuiTypography-root makeStyles-detailText-4 MuiTypography-body1"
                >
                  157
                </p>
              </td>
            </tr>
          </tbody>
        </table>
        <nav
          aria-labelledby="nested-list-subheader"
          className="MuiList-root makeStyles-root-11 makeStyles-sourceList-6 MuiList-padding"
        >
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root makeStyles-mainItem-12 MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemText-root makeStyles-listOpener-14"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Manga sources
              </span>
            </div>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
            <span
              className="MuiTouchRipple-root"
            />
          </div>
        </nav>
      </div>
    </div>
    <div
      className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <table
        aria-label="simple table"
        className="MuiTable-root"
        role={null}
      >
        <thead
          className="MuiTableHead-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head makeStyles-titleColumn-9"
              scope="col"
            >
              Title
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Chapter
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head makeStyles-releaseColumn-10"
              scope="col"
            >
              Released
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Group
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              <a
                className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href="https://mangaplus.shueisha.co.jp/viewer/1007322"
                onBlur={[Function]}
                onFocus={[Function]}
                style={
                  Object {
                    "textDecoration": "none",
                  }
                }
                target="_blank"
              >
                <span>
                  Z=157: Same Time, Same Place
                </span>
              </a>
            </th>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              157
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              2020-07-05T16:00:00.000Z
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              Shueisha
            </td>
          </tr>
          <tr
            className="MuiTableRow-root"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              <a
                className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href="https://mangadex.org/chapter/938629"
                onBlur={[Function]}
                onFocus={[Function]}
                style={
                  Object {
                    "textDecoration": "none",
                  }
                }
                target="_blank"
              >
                <span>
                  Z=156: Two Scientists
                </span>
              </a>
            </th>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              156
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              2020-06-26T17:01:59.000Z
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              MangaPlus
            </td>
          </tr>
          <tr
            className="MuiTableRow-root"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              <a
                className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href="https://mangaplus.shueisha.co.jp/viewer/1007024"
                onBlur={[Function]}
                onFocus={[Function]}
                style={
                  Object {
                    "textDecoration": "none",
                  }
                }
                target="_blank"
              >
                <span>
                  Z=156: Two Scientists
                </span>
              </a>
            </th>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              156
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              2020-06-26T16:00:00.000Z
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              Shueisha
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`Manga page should render correctly should render correctly when logged in 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg"
>
  <div
    className="MuiPaper-root makeStyles-paper-8 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <h4
      className="MuiTypography-root makeStyles-title-1 MuiTypography-h4"
    >
      Dr. STONE
    </h4>
    <div
      className="makeStyles-details-3"
    >
      <a
        href="https://myanimelist.net/manga/103897"
        rel="noreferrer noopener"
        target="_blank"
      >
        <img
          alt="Dr. STONE"
          className="makeStyles-thumbnail-2"
          src="https://mangadex.org/images/manga/20882.jpg?1585634146"
        />
      </a>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
      >
        <table
          className="makeStyles-infoTable-5"
        >
          <tbody>
            <tr>
              <td>
                <p
                  className="MuiTypography-root MuiTypography-body1"
                >
                  Latest release:
                </p>
              </td>
              <td>
                <p
                  aria-describedby={null}
                  className="MuiTypography-root makeStyles-detailText-4 MuiTypography-body1"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  title="Sun, 05 Jul 2020 16:00:00 GMT"
                >
                  2020-07-05T16:00:00.000Z
                </p>
              </td>
            </tr>
            <tr>
              <td>
                <p
                  className="MuiTypography-root MuiTypography-body1"
                >
                  Estimated release interval:
                </p>
              </td>
              <td>
                <p
                  className="MuiTypography-root makeStyles-detailText-4 MuiTypography-body1"
                >
                  7 days 0 hours
                </p>
              </td>
            </tr>
            <tr>
              <td>
                <p
                  className="MuiTypography-root MuiTypography-body1"
                >
                  Estimated next release:
                </p>
              </td>
              <td>
                <p
                  className="MuiTypography-root makeStyles-detailText-4 MuiTypography-body1"
                >
                  2020-07-12T16:00:00.000Z
                </p>
              </td>
            </tr>
            <tr>
              <td>
                <p
                  className="MuiTypography-root MuiTypography-body1"
                >
                  Latest chapter:
                </p>
              </td>
              <td>
                <p
                  className="MuiTypography-root makeStyles-detailText-4 MuiTypography-body1"
                >
                  157
                </p>
              </td>
            </tr>
          </tbody>
        </table>
        <nav
          aria-labelledby="nested-list-subheader"
          className="MuiList-root makeStyles-root-11 makeStyles-sourceList-6 MuiList-padding"
        >
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root makeStyles-mainItem-12 MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemText-root makeStyles-listOpener-14"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Manga sources
              </span>
            </div>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
            <span
              className="MuiTouchRipple-root"
            />
          </div>
        </nav>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-followButton-7 MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Follow
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
    <div
      className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <table
        aria-label="simple table"
        className="MuiTable-root"
        role={null}
      >
        <thead
          className="MuiTableHead-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head makeStyles-titleColumn-9"
              scope="col"
            >
              Title
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Chapter
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head makeStyles-releaseColumn-10"
              scope="col"
            >
              Released
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Group
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              <a
                className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href="https://mangaplus.shueisha.co.jp/viewer/1007322"
                onBlur={[Function]}
                onFocus={[Function]}
                style={
                  Object {
                    "textDecoration": "none",
                  }
                }
                target="_blank"
              >
                <span>
                  Z=157: Same Time, Same Place
                </span>
              </a>
            </th>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              157
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              2020-07-05T16:00:00.000Z
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              Shueisha
            </td>
          </tr>
          <tr
            className="MuiTableRow-root"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              <a
                className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href="https://mangadex.org/chapter/938629"
                onBlur={[Function]}
                onFocus={[Function]}
                style={
                  Object {
                    "textDecoration": "none",
                  }
                }
                target="_blank"
              >
                <span>
                  Z=156: Two Scientists
                </span>
              </a>
            </th>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              156
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              2020-06-26T17:01:59.000Z
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              MangaPlus
            </td>
          </tr>
          <tr
            className="MuiTableRow-root"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              <a
                className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href="https://mangaplus.shueisha.co.jp/viewer/1007024"
                onBlur={[Function]}
                onFocus={[Function]}
                style={
                  Object {
                    "textDecoration": "none",
                  }
                }
                target="_blank"
              >
                <span>
                  Z=156: Two Scientists
                </span>
              </a>
            </th>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              156
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              2020-06-26T16:00:00.000Z
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              Shueisha
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
